src/Middleware/IpFilter.php

```diff
@@ -27,7 +47,8 @@ public function __construct(string $allowedIp, ResponseFactoryInterface $respons
      */
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        if ($request->getServerParams()['REMOTE_ADDR'] !== $this->allowedIp) {
+        $networkResolver = $this->networkResolver->withServerRequest($request);
```

Comment:
`$this->networkResolver` seems to be optional so a check for `null` should be added with falling back to `$request->getServerParams()['REMOTE_ADDR']`. Alternatively it could be made mandatory.